### PR TITLE
Make installation and execution easier

### PR DESCRIPTION
**Pull request by jason0x43**
- `npm install` does everything
- squelch the compile errors
- add a run script

### CODE REVIEW

File: bin/dts-generator [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env node
+require('./dts-generator.js')
```
Comment (csnover) [inline bin/dts-generator]: What’s the benefit of this file?

File: tsextra.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/// <reference path="./typings/typescript/typescript.d.ts" />
+
+// These are declarations that exist on the typescript object but aren't currently included in the typescript.d.ts. This
+// file, and its reference in index.ts, may be removed if/when these functions are included in typescript.d.ts.
+
+declare module 'typescript' {
+	export function createEmitHostFromProgram(program: Program): any;
+	export function emitFiles(resolver: EmitResolver, host: any, targetSourceFile?: SourceFile): EmitResult;
+}
```
Comment (csnover) [inline tsextra.d.ts]: I’ve been informed these definitions exist in `typescript_internal.d.ts` so we can probably use that for the moment

Comment (jason0x43) [inline tsextra.d.ts]: They do not, at least not as of 2052ac.

File: bin/dts-generator (hunk 1 above)
Comment (jason0x43) [inline bin/dts-generator]: I prefer runnable files over typing `node xyz`.

File: tsextra.d.ts (hunk 2 above)
Comment (jason0x43) [inline tsextra.d.ts]: Well, emitFiles does not.

**Comment by csnover** [comment]
I just crammed a bunch of files into the repository directly for now. @DanielRosenwasser did an LKG update on the TypeScript repo today so I switched to use that as the dependency for now, but I’d like to move away from that sooner than later since it also pulls down all the tests and stuff which are huge and unnecessary. Hopefully they will get published nightlies soon. I tried to use @Arnavion’s repo in the interim but its `package.json` calls itself 'typescript-github' so when we try to install at 'typescript' (because all the typings are for `module 'typescript'`) npm will refuse to allow it (or at least refuses on my machine).

**Comment by Arnavion** [comment]
I didn't intend it to be a drop-in replacement for typescript w.r.t. the require statement. My repo is not the official TS repo so I was hesitant to have it call itself "typescript" in package.json. However I do see your point that the .d.ts files require you to import "typescript", so I will make that change.

Re: the missing _internal.d.ts you pointed out on IRC, I will add them.

**Comment by DanielRosenwasser** [comment]
We're hoping to put out something like a preview on npm. Not quite as frequent as a nightly (maybe every 1-3 weeks). If you really do need nightlies, you can log an issue and we can discuss it there.

**Pull request closed without merging**